modular_skyrat/modules/ashwalkers/code/effects/ash_rituals.dm

```diff
@@ -229,27 +234,106 @@
 
 /datum/ash_ritual/share_damage/ritual_success(obj/effect/ash_rune/success_rune)
 	. = ..()
-	//try to find the person to heal
+
 	var/mob/living/carbon/human/human_victim = locate() in get_turf(success_rune)
 	if(!human_victim)
 		return
-	//see how much damage they have
+
 	var/total_damage = human_victim.getBruteLoss() + human_victim.getFireLoss()
-	//see how many valid people to split the damage amongst
+
 	var/divide_damage = 0
 	var/list/valid_humans = list()
 	for(var/mob/living/carbon/human/human_share in range(2, get_turf(success_rune)))
-		//can't be the victim
 		if(human_share == human_victim)
 			continue
-		//can't be dead
 		if(human_share.stat == DEAD)
 			continue
 		valid_humans += human_share
 		divide_damage++
-	//how much damage each person will take
+
 	var/singular_damage = total_damage / divide_damage
-	//do the healing and damage taking
 	for(var/mob/living/carbon/human/human_target in valid_humans)
 		human_target.adjustBruteLoss(singular_damage)
 	human_victim.heal_overall_damage(human_victim.getBruteLoss(), human_victim.getFireLoss())
+
+/datum/ash_ritual/banish_kin
+	name = "Banish Kin"
+	desc = "Some kin are not fit for the tribe, this can solve that issue through democracy."
+	required_components = list()
+	consumed_components = list()
+
+/datum/ash_ritual/banish_kin/ritual_success(obj/effect/ash_rune/success_rune)
+	. = ..()
+	var/turf/src_turf = get_turf(success_rune)
+
+	var/mob/living/carbon/human/find_banished = locate() in src_turf
+	if(!find_banished)
+		return
+
+	var/list/responses = list(
+		"Yes" = 0,
+		"No" = 0,
+	)
+
+	var/list/asked_voters = list()
+
+	for(var/mob/living/carbon/human/poll_human in range(2, src_turf))
+		if(poll_human.stat != CONSCIOUS) //must be conscious
+			continue
+
+		if(!poll_human.mind.has_antag_datum(/datum/antagonist/ashwalker)) //must be an ashwalker
+			continue
+
+		asked_voters += poll_human
+
+	var/list/yes_voters = poll_candidates("Do you wish to banish [find_banished]?", poll_time = 10 SECONDS, group = asked_voters)
+
+	if(LAZYLEN(yes_voters) < LAZYLEN(asked_voters))
+		find_banished.balloon_alert_to_viewers("banishment failed!")
+		return
+
+	var/turf/teleport_turf = locate(rand(1,255), rand(1,255), success_rune.z)
+
+	while(!istype(teleport_turf, /turf/open/misc/asteroid/basalt))
```

Comment:
you can convert 295-298 into a `do while` loop, also why basalt

Reply:
That appears to be a copy-paste from `/datum/ash_ritual/incite_megafauna`'s ritual_success code, and it also occurs to me that both should have a `CHECK_TICK` or something as these loops can theoretically run indefinitely if they never hit an open tile.

Reply:
I don't see any examples of do while loops-- fixed the basalt thing.

Reply:
```
do
    code to while loop
 while(conditional)
 ```

Reply:
I just don't see why I need to convert it.

Reply:
Reducing repeat code

Reply:
It doesn't feel like reducing repeat code, it feels like just rewriting it in a fashion that is the same way?